lib/Doctrine/Deprecations/Deprecation.php

```diff
@@ -72,15 +70,49 @@ class Deprecation
      */
     public static function trigger(string $package, string $link, string $message, ...$args): void
     {
-        // Do not trigger this deprecation if it is temporarily ignored,
-        // because it is expected to be called for 1 or more times.
-        if (array_key_exists($link, self::$temporarilyIgnoredLinks)) {
-            self::$temporarilyIgnoredLinks[$link]--;
+        if (array_key_exists($link, self::$ignoredLinks)) {
+            self::$ignoredLinks[$link]++;
+        } else {
+            self::$ignoredLinks[$link] = 1;
+        }
+
+        if (self::$deduplication === true && self::$ignoredLinks[$link] > 1) {
+            return;
+        }
+
+        // do not move this condition to the top, because we still want to
+        // count occcurences of deprecations even when we are not logging them.
+        if (self::$type === self::TYPE_NONE) {
+            return;
+        }
+
+        if (isset(self::$ignoredPackages[$package])) {
+            return;
+        }
+
+        $backtrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 2);
+
+        $message = sprintf($message, ...$args);
+
+        self::delegateTriggerToBackend($message, $backtrace, $link, $package);
+    }
+
+    /**
+     * @param mixed $args
+     */
+    public static function triggerIfCalledFromOutside(string $package, string $link, string $message, ...$args): void
+    {
+        $backtrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 2);
```

Comment:
Unconditional usage of `debug_backtrace()` makes this method unnecessarily expensive even if logging is disabled.